content/reference/routing.md

```diff
@@ -0,0 +1,406 @@
+makes sense.
+
+Routing has a few parts to it. Routing must handle the browser's history API
+(e.g.  going forward & backwards). It must handle anchor tags, and programmatic
+actions. There's more than a few moving pieces.
+
+To perform routing, Choo uses a [Trie](https://en.wikipedia.org/wiki/Trie) data
+structure. This means that our routings is fast, and the order in which routes
+are added doesn't matter.
+
+_Note: It's recommended to read the [views]('/reference/views) chapter first, as
+we'll assume that you're already familiar with how views work. This chapter is
+intended to give an overview of how routing works in Choo._
+
+## Static routing
+Every application needs an entry point. Routes in Choo are defined relative to
+the host. The route `'/'` maps to `www.mysite.com`. The route `'/foo'` maps to
+`www.mysite.com/foo`.
+
+```js
+var html = require('choo/html')
+var choo = require('choo')
+
+var app = choo()                   // 1.
+app.route('/', view)               // 2.
+app.mount('body')                  // 3.
+
+function view () {                 // 4.
+  return html`
+    <body>Hello World</body>
+  `
+}
+```
+
+1. We need an instance of Choo to add our routes to, so let's create that
+   first.
+2. We're going to add a view on the `'/` route. This means that if people
+   navigate to `oursite.com`, this will be the route that is enabled.
+3. Now that we have our view, we can start rendering our application.
+4. We declare our view at the bottom of the page. Thanks to [scope
+   hoisting](https://developer.mozilla.org/en-US/docs/Glossary/Hoisting), we
+   can use it higher in the code. For now it doesn't really matter what's in
+   here, just that we return some DOM node.
+
+## Anchor tags
+There's no point in routing if you can't navigate between them. The easiest way
+to navigate between routes is to use `<a>` tags (anchor tags). Choo picks up
+whenever a tag was clicked, and figures out which route to trigger on the
+router.
+
+```js
+var html = require('choo/html')
+var choo = require('choo')
+
+var app = choo()
+app.route('/', view)               // 1.
+app.route('/second', second)       // 2.
+app.mount('body')                  // 3.
+
+function view () {
+  return html`
+    <body>
+      <a href="/second">
+        Navigate to the next route.
+      </a>
+    </body>
+  `
+}
+
+function second () {
+  return html`
+    <body>
+      <a href="/">
+        Navigate back.
+      </a>
+    </body>
+  `
+}
+```
+
+1. We define our base view on route `/`. This is the first route that's loaded
+   when a person visits our site. It contains a single anchor tag that points
+   to `/second`.
+2. We defined our second route as `/second`. This won't be shown unless someone
+   navigates to `/second`. When it's rendered, it contains a single anchor tag
+   that points to `/`.
+3. We render our app to the DOM. Once it's loaded, people can click on anchor
+   tags to switch between views.
+
```

Comment:
anchor tag is a bit specific. how about replacing it with link tag? more people have that association

Reply:
That's a pretty fair point, but it's hard to then distinguish from `<link>` tags, which are a thing we might / should touch on at some point?

If we follow [mdn](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/a), then they also call it an anchor tag there. Agree with you tho, but think it might cause trouble down the road later (perhaps).

Reply:
fair enough!